src/adapter/mysql.cr

```diff
@@ -5,6 +5,15 @@ require "mysql"
 class Granite::Adapter::Mysql < Granite::Adapter::Base
   QUOTING_CHAR = '`'
 
+  module Schema
+    TYPES = {
+      "AUTO_Int32" => "INT NOT NULL AUTO_INCREMENT PRIMARY KEY",
+      "AUTO_Int64" => "BIGINT NOT NULL AUTO_INCREMENT PRIMARY KEY",
```

Comment:
The capitalization here is unexpected. Why `AUTO_Int32`?

Reply:
Well, I'd like to resolve three kinds of **crystal objects** here.
1. Auto number (**value** + **metadata** like `Int32` or `Int64` + **this is autonumber**)
2. All crystal classes with module name (**value** like `Int32`, `Foo::Data`)
3. special field variables (**value** like `created_at` , `updated_at`)

Here, since they can be represented by **strings itself** for the case of 2 and 3, we can use simple  `Hash(String, String)` for the mapping dictionary.
Then, I faced a problem of how to embed the meta information of **autonumber** in the payload for the case 1.

Although there are various correct and complicated implementations, I prioritized the simplicity of implementation, and I embedded the meta information as "uppercase letters + _" in the string for the time being. 😸